[Crash Log] Mon, 14 Dec 2015 07:32:34 GMT

User:hybrtz <http://weibo.com/u/3295250915> 
Latest crash log: 
Android Version: 6.0
Device Model: Nexus 6
Device Manufacturer: motorola
App Version: 2.2.043
*********************
java.lang.RuntimeException: Unable to resume activity {us.shandian.blacklight/info.papdt.blacklight.ui.entry.UrlEntryActivity}: java.lang.IllegalStateException: Activity {us.shandian.blacklight/info.papdt.blacklight.ui.entry.UrlEntryActivity} did not call finish() prior to onResume() completing
	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:3103)
	at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:3134)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2481)
	at android.app.ActivityThread.-wrap11(ActivityThread.java)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1344)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:148)
	at android.app.ActivityThread.main(ActivityThread.java:5417)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
	at de.robv.android.xposed.XposedBridge.main(XposedBridge.java:114)
Caused by: java.lang.IllegalStateException: Activity {us.shandian.blacklight/info.papdt.blacklight.ui.entry.UrlEntryActivity} did not call finish() prior to onResume() completing
	at android.app.Activity.performResume(Activity.java:6324)
	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:3092)
	... 11 more

--- a/blacklight-base/src/main/java/info/papdt/blacklight/support/WeiboUrlUtility.java
+++ b/blacklight-base/src/main/java/info/papdt/blacklight/support/WeiboUrlUtility.java
@@ -23,7 +23,6 @@ import android.content.Context;
 import android.content.Intent;
 import android.net.Uri;
 import android.text.TextUtils;
-import android.widget.Toast;
 
 import info.papdt.blacklight.api.shorturl.ShortUrlApi;
 import info.papdt.blacklight.api.statuses.QueryIdApi;

--- a/blacklight-base/src/main/java/info/papdt/blacklight/ui/entry/UrlEntryActivity.java
+++ b/blacklight-base/src/main/java/info/papdt/blacklight/ui/entry/UrlEntryActivity.java
@@ -23,6 +23,7 @@ import android.app.Activity;
 import android.content.Context;
 import android.content.Intent;
 import android.net.Uri;
+import android.os.Build;
 import android.os.Bundle;
 import android.text.TextUtils;
 import android.widget.Toast;
@@ -47,6 +48,16 @@ public class UrlEntryActivity extends Activity
 			}
 		}
 		finish();
+	}
+
+	@Override
+	protected void onStart() {
+		super.onStart();
+
+		// workaround for M, see https://code.google.com/p/android-developer-preview/issues/detail?id=2353
+		if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+			setVisible(true);
+		}
 	}
 
 	private class UrlParseTask extends AsyncTask<Uri, Void, Intent> {

--- a/blacklight-base/src/main/java/info/papdt/blacklight/ui/entry/UrlEntryActivity.java
+++ b/blacklight-base/src/main/java/info/papdt/blacklight/ui/entry/UrlEntryActivity.java
@@ -20,13 +20,10 @@
 package info.papdt.blacklight.ui.entry;
 
 import android.app.Activity;
-import android.content.Context;
 import android.content.Intent;
 import android.net.Uri;
 import android.os.Build;
 import android.os.Bundle;
-import android.text.TextUtils;
-import android.widget.Toast;
 
 import info.papdt.blacklight.cache.login.LoginApiCache;
 import info.papdt.blacklight.support.AsyncTask;

